api/__tests__/routes/region.test.ts

```diff
@@ -0,0 +1,357 @@
+                                last_name: "De Superstudent",
+                                phone: "9876543210",
+                                student: false,
+                                super_student: true,
+                            },
+                            user_id: 2,
+                        },
+                    ],
+                },
+                { deleted: false, id: 3, name: "Region 3", users: [] },
+            ];
+
+            await runner.get({ url: "/region", expectedData: expected });
+        });
+
+        test("GET /region/:id", async () => {
+            const expected = [
+                { id: 3, name: "Region 3", deleted: false, users: [] },
+            ];
+
+            await runner.get({ url: "/region/3", expectedData: expected });
+        });
+
+        test("PATCH /region/:id", async () => {
+            const newRegion = {
+                id: 3,
+                name: "Updated Region 3",
+            };
+
+            const expectedRegion = {
+                id: 3,
+                name: "Updated Region 3",
+                deleted: false,
+                users: [],
+            };
+
+            await runner.patch({
+                url: "/region/3",
+                data: newRegion,
+                expectedResponse: expectedRegion,
+            });
+        });
+
+        test("DELETE /region/:id", async () => {
+            await runner.delete({ url: "/region/1" });
+
+            // verify that the region is truly deleted
+            const expected = [
+                {
+                    deleted: false,
+                    id: 2,
+                    name: "Region 2",
+                    users: [
+                        {
+                            id: 2,
+                            region_id: 2,
+                            user: {
+                                address_id: 2,
+                                admin: false,
+                                date_added: "2023-05-04T12:00:00.000Z",
+                                deleted: false,
+                                email: "superstudent@trottoir.be",
+                                first_name: "Toon",
+                                id: 2,
+                                last_login: "2023-05-04T12:00:00.000Z",
+                                last_name: "De Superstudent",
+                                phone: "9876543210",
+                                student: false,
+                                super_student: true,
+                            },
+                            user_id: 2,
+                        },
+                    ],
+                },
+                { deleted: false, id: 3, name: "Region 3", users: [] },
+            ];
+            await runner.get({
+                url: "/region",
+                expectedData: expected,
+            });
+        });
+
+        afterAll(() => {
+            app.close();
+        });
```

Comment:
Deze `afterAll` werkt enkel binnen de "Successful requests" blok. Deze moet in de buitenste blok komen.

Reply:
Er staat ook een `afterAll` in "Unsuccesful requests". Is het beter om maar 1 keer `afterAll` te doen in de buitenste block?

Reply:
Het gaat er niet om maar 1x te doen.
In de afterAll staat ``app.close()``. Nu blijkt dat de testen alsnog verder kunnen, blijkbaar dankzij de `requests` library, maar het is overbodig deze resource 2x vrij te geven. Het doet hier geen kwaad, maar we hoeven ons niet onnodig te herhalen